.github/workflows/cpp-packaging.yml

```diff
@@ -498,14 +576,33 @@ jobs:
     if: always()
     runs-on: ubuntu-latest
     steps:
-      - name: download artifact
+      - name: download SDK zip
         uses: actions/download-artifact@v2
         with:
-          name: firebase_cpp_sdk
+          name: firebase_cpp_sdk.zip
           path: .
+
+      - name: download hash
+        uses: actions/download-artifact@v2
+        with:
+          name: firebase_cpp_sdk_hash.txt
+          path: .
+
       - name: List binary SDK files.
         run: |
+          # Verify zipfile hash first.
+          ${{ env.hashCommand }} -c --quiet firebase_cpp_sdk_hash.txt
```

Comment:
FYI Github does not install sha256sum on MacOS, since MacOS ships with shasum. So I won't be able to include these hash-related commands in the test workflows (which run on all three OSes) as is. Though since they only log a warning, I'm assuming they're not essential.

Reply:
Yeah, we can get rid of that check from the general testing workflow, and move it to a standalone hash verification step that just runs on ubuntu, separately from everything else.

If the sha256sum command fails it will actually error (due to the command returning an error), though of the file_hashes.txt file is missing then *that* just a warning (since that file was not shipped in our most recent public release).

Reply:
Oh also, I believe shasum can be used to verify hashes created by sha256sum so I can experiment with that if you think we should keep this check. (It still won't do what we want on Windows, I think, though.)